framework/src/node/state_machine/block_context.ts

```diff
@@ -40,7 +40,7 @@ export class BlockContext {
 	private readonly _logger: Logger;
 	private readonly _eventQueue: EventQueue;
 	private readonly _header: BlockHeader;
-	private readonly _transactions?: ReadonlyArray<Transaction>;
+	private _transactions?: ReadonlyArray<Transaction>;
```

Comment:
Why we need to set the transactions after context intialization? I believe context must be always immutable object. If you really want to change the context must create new one with new transactions.

Reply:
`transactions`  can only be used in the `afterBlockExec` or `sealBlock` step now. That's why it's assigned after they been executed.